compiler/src/iree/compiler/Utils/ElementPackingUtils.cpp

```diff
@@ -114,14 +124,17 @@ Value calculateStorageElementCountInBytes(Location loc,
   if (needToPackSubByteElementBitWidth(elementBits)) {
     assert(8 % elementBits == 0);
     unsigned byteElements = 8 / elementBits;
-    // Perform some basic sanity check to make sure the total count is byte
-    // aligned for fully static shapes.
-    if (paddedDynamicDims.empty() && (staticCount * elementBits) % 8 != 0) {
-      return nullptr;
-    }
-    auto divisor = builder.create<arith::ConstantIndexOp>(loc, byteElements);
     // TODO(antiagainst): We may want to emit runtime check to make sure this is
     // divisible.
+    auto divisor = builder.create<arith::ConstantIndexOp>(loc, byteElements);
+    if (paddedDynamicDims.empty() && (staticCount * elementBits) % 8 != 0) {
+      if (clEnableI1Support)
+        value = builder.createOrFold<arith::CeilDivUIOp>(loc, value, divisor);
+      else
+        return nullptr;
```

Comment:
style nit: prefer early-exit style to save levels of nesting and simplify the code: https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code

```suggestion
      if (!clEnableI1Support) {
        return nullptr;
      }
      value = builder.createOrFold<arith::CeilDivUIOp>(loc, value, divisor);
```